### ISSUE TITLE
Extract statistics

### ISSUE THREAD

**Issue by Zomis** [Cardshifter/Cardshifter#121]
As we now have replays, it would be nice to be able to loop over a bunch of replays and extract some statistics from them. Example statistics include:
- Cards played (separated into creatures and enchantments)
- Cards scrapped, scrap got, mana used
- Turns played
- Biggest minion on the battlefield

Feel free to add more statistics you'd like to know about!

**Comment by Phrancis** [comment]
I think all kinds of interesting stats could be derived from game- and player-level data, especially if we can persist date-time data.

On the other hand, as much as I would like card-level data (e.g. which cards deal most damage, most often part of winning deck, most popular, etc.) would be too subjective for right now, as cards are highly subject to change over time.

**Comment by Zomis** [comment]
I think the fact that they are highly subject to change over time is exactly the reason why we should gather statistics for that, so that we can extract more information about when and how to change a card.

When extracting the statistics, it would be run on several specific replay files. We can choose what replay files to run it on. We would only run it on one version of the mod at a time, of course.

**Comment by Phrancis** [comment]
In that light I can see that, as long as it is versioned per card/mod then it would be good information. I was more thinking of "cumulative" information I guess, as far as games/players are concerned. Probably wouldn't be too difficult to do both cumulative and versioned. Planning on just running replay data as needed through Java or using a DB for these stats?

**Comment by Zomis** [comment]
Replay data will be just ran through Java. Possibly exported to some nice format though.

Cumulative results for player wins and stuff would of course have to be persisted to a DB, but that information will not be nearly as detailed as what this issue is about.

**Comment by Phrancis** [comment]
A nice format would be cool. Would it be possible to render either a CSV or Excel file so the data is easy to filter/manipulate/aggregate?

**Comment by Zomis** [comment]
Depends a bit on what information I retrieve from this, and how I will structure it. Many things are still unclear but I will keep your suggestions in mind.

**Comment by Phrancis** [comment]
I think this will in part be improved with the database we are setting up?

**Comment by Zomis** [comment]
Yes, but as the database *does not contain game logic*, things like "total amount of damage dealt in a game" is not something that the database can possibly handle.
